### ISSUE TITLE
[BUG]: Model with a field called "container" will throw an exception on serialize 

### ISSUE THREAD

**Issue by Nialld** [phalcon/cphalcon#14755]
**Describe the bug**
Model with a field called "container" will throw an exception on serialize 

**To Reproduce**

CREATE TABLE `test` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `container` varchar(64) DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB

and use phalcon dev tools to create a model of this table

create an instance of the model and serialize

im seeing error "Serialization of 'Closure' is not allowed" 
OR 
"Trying to call method getshared on a non-object" if container is set to empty string

Phalcon - 4.0.2
PHP - 7.2.24
OS Ubuntu 
Install from repo
Nginx

https://github.com/phalcon/cphalcon/blob/c4a804a70c9b89728f010b23935f450c79690bd7/phalcon/Mvc/Model.zep#L1676 appears to be the offending line

**Comment by Jurigag** [comment]
@ruudboon not sure if this actually fixeable. If yes then we would need to change all properties names, also didn't we have names like `_property` in phalcon 3? I always though that this was one of the main reason - to avoid cases like this.

**Comment by niden** [comment]
This is definitely not fixable. The name `container` is a protected property used for the DI container inside the model. We used to have something in the lines of `_dependencyInjector` which was kind of difficult to collide with but that was changed following the coding standard we agreed upon for v4. 

This of course all comes from the magic methods and the lack of a real property/field map in the models. We _could_ fix this (and I have not tested this) if we create an array store inside each model which will hold all the fields. That would require a significant amount of refactoring on the models and the database layer.

I am not sure if changing the name of the field in the model by using the columnMap would help. Failing that you will need to rename the field in the database and subsequently the model
